plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -235,100 +224,72 @@ class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
 
         val job = coroutineScope.launch {
             try {
-                val responseIterable = CodeWhispererClientAdaptor.getInstance(requestContext.project).generateCompletionsPaginator(
-                    buildCodeWhispererRequest(
-                        requestContext.fileContextInfo,
-                        requestContext.awaitSupplementalContext(),
-                        requestContext.customizationArn,
-                        requestContext.profileArn,
-                        requestContext.workspaceId,
-                    )
-                )
-
                 var startTime = System.nanoTime()
-                requestContext.latencyContext.codewhispererPreprocessingEnd = System.nanoTime()
-                requestContext.latencyContext.paginationAllCompletionsStart = System.nanoTime()
                 CodeWhispererInvocationStatus.getInstance().setInvocationStart()
-                var requestCount = 0
-                for (response in responseIterable) {
-                    requestCount++
-                    val endTime = System.nanoTime()
-                    val latency = TimeUnit.NANOSECONDS.toMillis(endTime - startTime).toDouble()
-                    startTime = endTime
-                    val requestId = response.responseMetadata().requestId()
-                    val sessionId = response.sdkHttpResponse().headers().getOrDefault(KET_SESSION_ID, listOf(requestId))[0]
-                    if (requestCount == 1) {
-                        requestContext.latencyContext.codewhispererPostprocessingStart = System.nanoTime()
-                        requestContext.latencyContext.paginationFirstCompletionTime =
-                            (endTime - requestContext.latencyContext.codewhispererEndToEndStart).toDouble()
-                        requestContext.latencyContext.firstRequestId = requestId
-                        CodeWhispererInvocationStatus.getInstance().setInvocationSessionId(sessionId)
-                    }
-                    if (response.nextToken().isEmpty()) {
-                        requestContext.latencyContext.paginationAllCompletionsEnd = System.nanoTime()
+                var nextToken: Either<String, Int>? = null
+                do {
+                    val result = AmazonQLspService.executeIfRunning(requestContext.project) { server ->
```

Comment:
need to handle case where server is not running

Reply:
will handle jsonRpcException in catch